pyccel/parser/semantic.py

```diff
@@ -439,10 +434,11 @@ def _infere_type(self, expr, **settings):
             d_var['shape'         ] = expr.shape
             d_var['rank'          ] = expr.rank
             d_var['order'         ] = expr.order
-            d_var['is_pointer'    ] = False
-            d_var['allocatable'   ] = any(getattr(a, 'allocatable', False) for a in expr.args)
-            d_var['is_stack_array'] = not d_var['allocatable']
             d_var['cls_base'      ] = TupleClass
+            if any(getattr(a, 'memory_handling', False) and a.memory_handling == 'heap' for a in expr.args):
```

Comment:
The second argument to `getattr` is the default value so this can be written more simply as:
```
if any(getattr(a, 'memory_handling', 'none') == 'heap' for a in expr.args)
```
(Or it can be written as before with a more logical default value if the properties are added)